src/satellite/process.ts

```diff
@@ -418,12 +423,38 @@ export class SatelliteProcess implements Satellite {
   }
 
   async _getMeta(key: string): Promise<string> {
+    const rows = await this._getMetaValue(key)
+    return rows[0]!.value as any
+  }
+
+  private async _getMetaValue(key: string): Promise<Row[]> {
```

Comment:
Can we keep just _getMeta or _getMetaValue?
The _getMeta was a bit lazy in the sense that it was not handling types properly. This is historical: before meta values could only be strings.

I think the interface should return only value or null (or undefined), but never a row array. It would be equally easy to build the getClientId around that interface without exposing the row array.